### DISCUSSION TITLE
Manual route calculation after route changes

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#449] [Q&A] [answered]
**Manual route calculation** disables automatic route calculation (as we know it now).

Add / move / delete waypoints, change the route without waiting for it to be calculated with each change.
The undo button cancels all pending route changes and resets the waypoints to the last displayed route.

When we complete the route changes, we can select "Recalculate route" to see the new calculated route.
We can also have a screen button to inform about existing changes and route calculation waiting to run.

---

We can add it as an extra route calculation mode option:

- https://github.com/devemux86/cruiser/discussions/142 (new default)

- **Manual route calculation** after completing all route changes

<img src="https://github.com/devemux86/cruiser/assets/3484020/5babf672-86fa-4895-a5c0-196bf9b17f01" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/9623dcba-d65d-45e7-b995-0c303f1a1d1c" width="50%"/>

**Comment by Leeb64** [comment]
Very good idea!
If you plan to set several intermediate points anyway, the unnecessary and long route calculation is annoying.

**Comment by devemux86** [comment]
**Manual route calculation** option after route planning is complete is available in:
- https://github.com/devemux86/cruiser/discussions/451

Please test it well to see if everything works properly. Instructions in the announcement above: https://github.com/devemux86/cruiser/discussions/449

If you see something strange, please report the route and detailed steps so we can reproduce it.

**Reply by bios4** [reply]
Very nice feature - works flawlessly with a (very short) test route from "my places". 👍

EDIT:

It works flawlessly when shuffling around EXISTING way-/via points.
When adding NEW way-/via-points in between the existing ones, the newly added point gets the very last number of all points, but not an "in between number" of the previous and next existing point!

**Comment by devemux86** [comment]
I found an issue - hopefully you can reproduce it ...
Switching to manual route calculation inserting a waypoint (regardless if shaping or via point) takes the number of the end waypoint. Intelligent route calculation works fine. Here the screenshots:
<img height="200" src="https://github.com/devemux86/cruiser/assets/121806238/efe5a7f0-b6de-445a-8432-6cb2a42843b3"/>  <img height="200" src="https://github.com/devemux86/cruiser/assets/121806238/24b44046-e68e-42a1-9155-a5b4a1ca5d82"/>  <img height="200" src="https://github.com/devemux86/cruiser/assets/121806238/051de409-b8d6-43f2-937b-248db8ab66a4"/>  <img height="200" src="https://github.com/devemux86/cruiser/assets/121806238/0c6aa19d-ebbb-408f-9d09-745501dffeb1"/>  <img height="200" src="https://github.com/devemux86/cruiser/assets/121806238/3c277acf-56f1-46f6-a2d2-326af417ede2"/>
This is the imported file: [Cruiser Day5 - ACT PT.zip](https://github.com/devemux86/cruiser/files/14359250/Cruiser.Day5.-.ACT.PT.zip)

_Originally posted by @WolfgangZ42 in https://github.com/devemux86/cruiser/discussions/142#discussioncomment-8542808._

**Reply by devemux86** [reply]
@WolfgangZ42

> Switching to manual route calculation inserting a waypoint (regardless if shaping or via point) takes the number of the end waypoint.

Indeed this currently happens. A bit complicated because in manual mode,
new points are really attached to the route after we press "recalculate".

I will need to check how we can improve this. :slightly_smiling_face:

**Reply by devemux86** [reply]
Order of via / shaping points in manual route calculation will work properly in the next update.

Edit: it is available.

**Comment by devemux86** [comment, marked as answer]
Manual route calculation after route changes is available in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
